tests/test_coreg.py

```diff
@@ -911,3 +1027,241 @@ def test_warp_dem() -> None:
         plt.subplot(144)
         plt.imshow(dem - untransformed_dem, cmap="coolwarm_r", vmin=-10, vmax=10)
         plt.show()
+
+
+def test_create_inlier_mask() -> None:
+    """Test that the create_inlier_mask function works expectedly."""
+    warnings.simplefilter("error")
+
+    ref, tba, outlines = load_examples()  # Load example reference, to-be-aligned and outlines
+
+    # - Assert that without filtering create_inlier_mask behaves as if calling Vector.create_mask - #
+    # Masking inside - using Vector
+    inlier_mask_comp = ~outlines.create_mask(ref)
+    inlier_mask = xdem.coreg.create_inlier_mask(
+        tba,
+        ref,
+        [
+            outlines,
+        ],
+        filtering=False,
+    )
+    assert np.all(inlier_mask_comp == inlier_mask)
+
+    # Masking inside - using string
+    inlier_mask = xdem.coreg.create_inlier_mask(
+        tba,
+        ref,
+        [
+            outlines.name,
+        ],
+        filtering=False,
+    )
+    assert np.all(inlier_mask_comp == inlier_mask)
+
+    # Masking outside - using Vector
+    inlier_mask_comp = outlines.create_mask(ref)
+    inlier_mask = xdem.coreg.create_inlier_mask(
+        tba,
+        ref,
+        [
+            outlines,
+        ],
+        inout=[
+            -1,
+        ],
+        filtering=False,
+    )
+    assert np.all(inlier_mask_comp == inlier_mask)
```

Comment:
Here you could just compare with `~inlier_mask_comp` without re-generating it above after the start of the "Masking outside" case, I think

Reply:
Done in [88391b1](https://github.com/GlacioHack/xdem/pull/329/commits/88391b18ee18c0ff67e4a2c2f1ffa4e226a2a81c)